queries.sql

```diff
@@ -9,3 +9,57 @@ SELECT * FROM animals WHERE neutered = true;
 SELECT * FROM animals WHERE name <> 'Gabumon';
 SELECT * FROM animals WHERE weight_kg BETWEEN 10.4 AND 17.3;
 
+BEGIN; -- start the transaction
+UPDATE animals SET species = 'unspecified'; -- set the species column to 'unspecified' for all rows
+SELECT * FROM animals; -- verify that the change was made
+ROLLBACK; -- roll back the transaction
+SELECT * FROM animals; -- verify that the species column went back to its original state
+
+BEGIN; -- start the transaction
+
+UPDATE animals SET species = 'digimon' WHERE name LIKE '%mon'; -- set the species column to 'digimon' for all animals with name ending in 'mon'
+UPDATE animals SET species = 'pokemon' WHERE species IS NULL; -- set the species column to 'pokemon' for all animals without a species
+
+SELECT * FROM animals; -- verify the changes within the transaction
+
+COMMIT; -- commit the transaction
+
+BEGIN; -- start the transaction
+
+DELETE FROM animals; -- delete all records from the animals table
+
+SELECT COUNT(*) FROM animals; -- verify that all records have been deleted within the transaction
+
+ROLLBACK; -- roll back the transaction
+
+SELECT COUNT(*) FROM animals; -- verify that all records are still present after the rollback
+
+BEGIN; -- start the transaction
+
+DELETE FROM animals WHERE date_of_birth > '2022-01-01'; -- delete all animals born after Jan 1st, 2022
+
+SAVEPOINT my_savepoint; -- create a savepoint for the transaction
+
+UPDATE animals SET weight_kg = weight_kg * -1; -- update all animals' weight to be their weight multiplied by -1
+
+ROLLBACK TO my_savepoint; -- rollback the transaction to the savepoint
+
+UPDATE animals SET weight_kg = weight_kg * -1 WHERE weight_kg < 0; -- update all animals' weights that are negative to be their weight multiplied by -1
+
+COMMIT; -- commit the transaction
+
+SELECT COUNT(*) FROM animals;
+SELECT COUNT(*) FROM animals WHERE escape_attempts = 0;
+SELECT AVG(weight_kg) FROM animals;
+SELECT neutered, SUM(escape_attempts) as total_escapes
+FROM animals
+GROUP BY neutered
+ORDER BY total_escapes DESC
+LIMIT 1;
```

Comment:
- [ ] Kindly make sure this query answers correctly to the question asked. You are supposed to get animals with the most escape_attempts, neutered or not neutered animals, and not the sum of escape_attempts on each type. To fix the issue you can just edit your query where you use `SUM(escape_attempts)` to have `MAX(escape_attempts)` or `AVG(escape_attempts)`(both of these approaches will give you the same result).

- [ ] Once you make the change above make sure you take the screenshot result and update the screenshot from the pull request description with the correct one. Good job and happy coding 🎉